src/databricks/labs/ucx/source_code/astlinter.py

```diff
@@ -0,0 +1,58 @@
+import ast
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MatchingVisitor(ast.NodeVisitor):
+
+    def __init__(self, node_type: type, match_nodes: list[tuple[str, type]]):
+        self.matched_nodes: list[ast.AST] = []
```

Comment:
if something is public, let's expose this as a method - would be easier to refactor.

Reply:
done